data/transactions/logic/eval_test.go

```diff
@@ -110,6 +110,40 @@ func TestEmptyProgram(t *testing.T) {
 	require.False(t, pass)
 }
 
+// TestMinTealVersionParamEval tests eval/check reading the MinTealVersion from the param
+func TestMinTealVersionParamEvalCheck(t *testing.T) {
+	t.Parallel()
+	params := defaultEvalParams(nil, nil)
+	version2 := uint64(rekeyingEnabledVersion)
+	params.MinTealVersion = &version2
+	program := make([]byte, binary.MaxVarintLen64)
+	// set the teal program version to 1
+	binary.PutUvarint(program, 1)
+
+	_, err := Check(program, params)
+	require.Contains(t, err.Error(), fmt.Sprintf("program version must be >= %d", appsEnabledVersion))
+
+	// If the param is read correctly, the eval should fail
+	pass, err := Eval(program, params)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), fmt.Sprintf("program version must be >= %d", appsEnabledVersion))
+	require.False(t, pass)
+}
+
+func TestTxnFieldToTealValue(t *testing.T) {
+
+	txn := transactions.Transaction{}
+	value := uint64(9999)
```

Comment:
nit: the test is not wrong, but it's a classic for testing several key values : 0, 1, 2, 2^32-1, 2^64-1

Reply:
added the test for the extra values.